lib/stdio.js

```diff
@@ -39,3 +39,23 @@ module.exports = opts => {
 
 	return result;
 };
+
+module.exports = stdio;
+
+module.exports.fork = opts => {
+	const defaultOption = (opts && opts.silent && !opts.stdio) ? 'pipe' : 'inherit';
+
+	let stdioOption = stdio(opts || {stdio: defaultOption});
+
+	if (typeof stdioOption === 'string') {
+		stdioOption = [...new Array(3)].fill(stdioOption);
+	} else if (Array.isArray(stdioOption)) {
+		stdioOption = stdioOption.map(channel => channel === undefined ? defaultOption : channel);
+	}
+
+	if (!stdioOption.includes('ipc')) {
```

Comment:
[Node.js code](https://github.com/nodejs/node/blob/0ca9297c3945f52325fcd08afe3fb9dfe3adf123/lib/internal/child_process.js#L228) pushes `ipc` as the last element, even if it's already present.

Reply:
True only if [`options.stdio` is not an array](https://github.com/nodejs/node/blob/master/lib/child_process.js#L97)

Reply:
No it's not, check the lines above:

```js
  if (typeof options.stdio === 'string') {
    options.stdio = stdioStringToArray(options.stdio, 'ipc');
```

Reply:
In my case, when the user define a string, `stdio.fork` append an `ipc` channel at the end of length 3 array. Check [tests](https://github.com/sindresorhus/execa/pull/200/files#diff-ddb2f51fa9f852e91d150fdec4404ba8R60) for more example.